anndata/_core/merge.py

```diff
@@ -107,6 +109,19 @@ def equal_dataframe(a, b) -> bool:
     return a.equals(b)
 
 
+@equal.register(DaskArray)
+def equal_dask_array(a, b) -> bool:
+    import dask.array as da
+
+    # TODO: Add potential performance warning
+    # TODO: In which cases do we test the else case?
+    # Why don't we use these dispatches in assert_array_equal anyways?
+    if not isinstance(b, DaskArray):
+        raise NotImplementedError()
+        # TODO: is it better to convert b to DaskArray here?
+    return da.equal(a, b, where=~(da.isnan(a) == da.isnan(b))).all()
```

Comment:
@ivirshup, should I focus on doing this the way xarray did? or should I give an error or warning?